easybuild/easyblocks/a/aocc.py

```diff
@@ -0,0 +1,125 @@
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_AOCC(PackedBinary):
+    """
+    Support for installing the AOCC compilers
+    """
+
+    @staticmethod
+    def extra_options():
+        extra_vars = {
+            'clangversion':      [None, "Clang Version on which AOCC is based on (10.0.0 or 11.0.0 or ...)", CUSTOM],
+        }
+        return PackedBinary.extra_options(extra_vars)
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor, define custom class variables specific to AOCC."""
+        super(EB_AOCC, self).__init__(*args, **kwargs)
+
+        self.clangversion = self.cfg['clangversion']
+
+    def _aocc_guess_clang_version(self):
+        map_aocc_to_clang_ver = {
+            '2.3.0': '11.0.0',
+        }
+
+        if self.version in map_aocc_to_clang_ver:
+            return map_aocc_to_clang_ver[self.version]
+        else:
+            error_lines = [
+                "AOCC is based on Clang. Guessing Clang version in easyblock failed.",
+                "You should either:",
+                "- specify `clangversion` in the easyconfig;",
+                "- extend `map_aocc_to_clang_ver` in the easyblock;",
+            ]
+            raise EasyBuildError('\n'.join(error_lines))
+
+    def prepare_step(self, *args, **kwargs):
+        super(EB_AOCC, self).prepare_step(*args, **kwargs)
+
+        # EULA for AOCC must be accepted via --accept-eula EasyBuild configuration option,
+        # or via 'accept_eula = True' in easyconfig file
+        self.check_accepted_eula(more_info='http://developer.amd.com/wordpress/media/files/AOCC_EULA.pdf')
+
+        # AOCC is based on Clang. Try to guess the clangversion from the AOCC version
+        # if clangversion is not specified in the easyconfig
+        if self.clangversion is None:
+            self.clangversion = self._aocc_guess_clang_version()
+
+    def sanity_check_step(self):
+        """Custom sanity check for AOCC, based on sanity check for Clang."""
+        shlib_ext = get_shared_lib_ext()
+        custom_paths = {
+            'files': [
+                "bin/clang", "bin/clang++", "bin/flang", "bin/llvm-ar", "bin/llvm-nm", "bin/llvm-as", "bin/opt",
+                "bin/llvm-link", "bin/llvm-config", "bin/llvm-symbolizer", "include/llvm-c/Core.h",
+                "include/clang-c/Index.h", "lib/libclang.%s" % shlib_ext,
+                "lib/clang/%s/include/stddef.h" % self.clangversion, "bin/scan-build", "bin/scan-view",
+                "bin/lld",
+                "lib/libc++.%s" % shlib_ext, "lib/libc++abi.%s" % shlib_ext,
+                "lib/libomp.%s" % shlib_ext, "lib/clang/%s/include/omp.h" % self.clangversion,
+            ],
+            'dirs': ["bin", "lib", "lib32", "include/clang", "include/llvm", "lib/clang/%s/lib" % self.clangversion],
```

Comment:
There's no point in checking for directories if we're already checking for specific files in those directories, so please remove `bin`, `lib`, `include/clang`, and also sort the remaining entries alphabetically.